cli/src/validate.rs

```diff
@@ -203,27 +203,41 @@ fn print_validation_errors(
     let mut num_suboptimal_reports: usize = 0;
     for report_validation in report_validations {
         let num_test_cases = report_validation.0.test_cases_flat().len();
-        let num_invalid_validation_errors = report_validation
+
+        let num_report_validation_errors =
+            report_validation.0.report_invalid_validation_issues().len();
+        let num_test_suite_validation_errors = report_validation
             .0
             .test_suite_invalid_validation_issues_flat()
-            .len()
-            + report_validation
-                .0
-                .test_case_invalid_validation_issues_flat()
-                .len();
-        let num_validation_warnings = report_validation
+            .len();
+        let num_test_case_validation_errors = report_validation
+            .0
+            .test_case_invalid_validation_issues_flat()
+            .len();
+        let num_validation_errors = num_report_validation_errors
+            + num_test_suite_validation_errors
+            + num_test_case_validation_errors;
+
+        let num_report_validation_warnings = report_validation
+            .0
+            .report_suboptimal_validation_issues()
+            .len();
+        let num_test_suite_validation_warnings = report_validation
             .0
             .test_suite_suboptimal_validation_issues_flat()
-            .len()
-            + report_validation
-                .0
-                .test_case_suboptimal_validation_issues_flat()
-                .len();
+            .len();
+        let num_test_case_validation_warnings = report_validation
+            .0
+            .test_case_suboptimal_validation_issues_flat()
+            .len();
+        let num_validation_warnings = num_report_validation_warnings
+            + num_test_suite_validation_warnings
+            + num_test_case_validation_warnings;
 
-        let num_validation_errors_str = if num_invalid_validation_errors > 0 {
-            num_invalid_validation_errors.to_string().red()
+        let num_validation_errors_str = if num_validation_errors > 0 {
+            num_validation_errors.to_string().red()
         } else {
-            num_invalid_validation_errors.to_string().green()
+            num_validation_errors.to_string().green()
         };
```

Comment:
Considering this and the additional logic in the follow up PR, it might be a good idea to move some of this logic into the `context` crate since it can be used elsewhere to get quick totals for validation issues.

Reply:
Done 👍 should look a lot nicer now